apps/ledger-live-mobile/src/newArch/features/WalletSync/screens/Activation/ActivationDrawer.tsx

```diff
@@ -1,37 +1,57 @@
-import React from "react";
+import React, { useState } from "react";
 import QueuedDrawer from "LLM/components/QueuedDrawer";
-import Activation from "../../components/Activation";
 import { TrackScreen } from "~/analytics";
-import ChooseSyncMethod from "../Synchronize/ChooseMethod";
+import { useWindowDimensions } from "react-native";
+import { Flex } from "@ledgerhq/native-ui";
+import ActivationFlow from "../../components/Activation/ActivationFlow";
+import { Steps } from "../../types/Activation";
+import DrawerHeader from "../../components/Synchronize/DrawerHeader";
 
 type Props = {
   isOpen: boolean;
-  reopenDrawer: () => void;
+  startingStep: Steps;
   handleClose: () => void;
 };
 
-const ActivationDrawer = ({ isOpen, handleClose, reopenDrawer }: Props) => {
-  const [isSyncMethodDrawerOpen, setIsSyncMethodDrawerOpen] = React.useState(false);
+const ActivationDrawer = ({ isOpen, startingStep, handleClose }: Props) => {
+  const [currentStep, setCurrentStep] = useState<Steps>(startingStep);
+  const { height } = useWindowDimensions();
+  const maxDrawerHeight = height - 180;
 
-  const onPressCloseDrawer = () => {
-    setIsSyncMethodDrawerOpen(false);
-    reopenDrawer();
-  };
+  const CustomDrawerHeader = () => <DrawerHeader onClose={handleClose} />;
+
+  const handleStepChange = (step: Steps) => setCurrentStep(step);
+
+  let goBackCallback: () => void;
+
+  const hasCustomHeader = currentStep === Steps.QrCodeMethod && startingStep === Steps.Activation;
 
-  const openSyncMethodDrawer = () => {
-    setIsSyncMethodDrawerOpen(true);
+  const canGoBack = currentStep === Steps.ChooseSyncMethod && startingStep === Steps.Activation;
+
+  const resetStep = () => setCurrentStep(startingStep);
+
+  const onClose = () => {
+    resetStep();
     handleClose();
   };
 
   return (
     <>
       <TrackScreen />
-      <QueuedDrawer isRequestingToBeOpened={isOpen} onClose={handleClose}>
-        <Activation isInsideDrawer openSyncMethodDrawer={openSyncMethodDrawer} />
-      </QueuedDrawer>
-
-      <QueuedDrawer isRequestingToBeOpened={isSyncMethodDrawerOpen} onClose={onPressCloseDrawer}>
-        <ChooseSyncMethod />
+      <QueuedDrawer
+        isRequestingToBeOpened={isOpen}
+        onClose={onClose}
+        CustomHeader={hasCustomHeader ? CustomDrawerHeader : undefined}
+        hasBackButton={canGoBack}
+        onBack={() => goBackCallback()}
+      >
+        <Flex maxHeight={maxDrawerHeight}>
```

Comment:
You just have to do maxHeight={"100%"} it will works

Reply:
It's taking the all the height screen with 100% and a custom header making it not closable